### PR TITLE
Write the catchpoint file in parallel of reading from the database

### PR DESCRIPTION
**Pull request by tsachiherman**
## Summary

This PR improves the speed of the catchpoint file writer; it parallelize the gzip file writing with the account database reading.

## Test Plan

Existing tests already cover the functional portion.

## Performance Gains

| Number of accounts   |     Before the change      |  With the change | Performance gains |
|:----------:|:-------------:|:------:|:------:|
| 6,000,000 |  22 seconds | 13 seconds | +90% |

### CODE REVIEW

**Review by algonautshant — APPROVED** [review]
Looks good. 
I just have one comment.

File: ledger/catchpointwriter_test.go [hunk 1]
```diff
@@ -217,7 +217,7 @@ func TestBasicCatchpointWriter(t *testing.T) {
 			require.NoError(t, err)
 			require.Equal(t, uint64(len(accts)), uint64(len(balances.Balances)))
 		} else {
-			require.Failf(t, "unexpected tar chunk name %s", header.Name)
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply ledger/catchpointwriter_test.go:220]: Correct. It is unexpected.
Unfortunately, that's how the `require` package is written, and this change just align with a "correct" implementation.
i.e.
```
// Failf reports a failure through
func Failf(t TestingT, failureMessage string, msg string, args ...interface{}) bool {
	if h, ok := t.(tHelper); ok {
		h.Helper()
	}
	return Fail(t, failureMessage, append([]interface{}{msg}, args...)...)
}
```

File: ledger/catchpointwriter.go [hunk 2]
```diff
@@ -188,43 +206,84 @@ func (cw *catchpointWriter) WriteStep(stepCtx context.Context) (more bool, err e
...
 				}
-				cw.writtenBytes = fileInfo.Size()
-				return false, nil
 			}
 			cw.balancesChunk.Balances = nil
 		}
 	}
 }
 
+func (cw *catchpointWriter) asyncWriter(balances chan catchpointFileBalancesChunk, response chan error, initialBalancesChunkNum uint64) {
```
[the remark this answers is not in the document]
Comment (tsachiherman) [reply ledger/catchpointwriter.go:245]: unless @brianolson would have any notes, I'll be making this change on a subsequent PR

**Comment by tsachiherman** [comment]
Optimize the catchpoint file writer performance.

**Pull request merged**